addon/components/power-select-typeahead/trigger.js

```diff
@@ -45,48 +57,67 @@ export default Component.extend({
       } else {
         run.schedule('actions', null, newSelect.actions.open);
       }
-    } else if (!isBlank(newSelect.lastSearchedText) && newSelect.options.length === 0 && newSelect.loading) {
-      run.schedule('actions', null, newSelect.actions.close, null, true);
-    } else if (oldSelect.loading && !newSelect.loading && newSelect.options.length > 0) {
-      run.schedule('actions', null, newSelect.actions.open);
     }
   },
 
-  // Actions
   actions: {
+    /**
+     * on mousedown prevent propagation of event
+     *
+     * @private
+     * @method stopPropagation
+     * @param {Object} event
+     */
     stopPropagation(e) {
       e.stopPropagation();
     },
 
+    /**
+     * called from power-select internals
+     *
+     * @private
+     * @method handleKeydown
+     * @param {Object} event
+     */
     handleKeydown(e) {
+      // up or down arrow and if not open, no-op and prevent parent handlers from being notified
       if ([38, 40].indexOf(e.keyCode) > -1 && !this.get('select.isOpen')) {
         e.stopPropagation();
         return;
       }
       let isLetter = e.keyCode >= 48 && e.keyCode <= 90 || e.keyCode === 32; // Keys 0-9, a-z or SPACE
+      // if isLetter, escape or enter, prevent parent handlers from being notified
       if (isLetter || [13, 27].indexOf(e.keyCode) > -1) {
+        let select = this.get('select');
+        // open if loading msg configured
+        if (!select.isOpen) {
+          run.schedule('actions', null, select.actions.open);
```

Comment:
@cibernox the dropdown should only open right away to show loadingMessage if loadingMessage is configured. However, at this level (the trigger), I am not sure how to tell if a loadingMessage was configured or not. I have a failing test below.

Any thoughts on how to tell if the loadingMessage was set?

Reply:
You're right, there is no way to know that from here. Perhaps it could be part of the `publicAPI` (https://github.com/cibernox/ember-power-select/blob/master/addon/components/power-select.js#L53-L65)

Or perhaps we could just pass it to the trigger here: https://github.com/cibernox/ember-power-select/blob/master/addon/templates/components/power-select.hbs#L29-L44

Either seems fine. I'd probably start with the second approach.